src/test/java/org/apache/commons/io/output/ByteArrayOutputStreamTest.java

```diff
@@ -198,7 +205,23 @@ void testToBufferedInputStreamEmpty(final String baosName, final IOFunction<Inpu
 
             try (InputStream buffered = toBufferedInputStreamFunction.apply(bain)) {
                 assertEquals(0, buffered.available());
+            }
+        }
+    }
+
+    @ParameterizedTest(name = "[{index}] {0}")
+    @MethodSource("toBufferedInputStreamFunctionFactories")
+    void testToBufferedInputStreamEmptyFile(final String baosName, final IOFunction<InputStream, InputStream> toBufferedInputStreamFunction,
+                                            final @TempDir Path temporaryFolder) throws IOException {
+        final Path emptyFile = Files.createTempFile(temporaryFolder, getClass().getSimpleName(), "-empty.txt");
+
+        try (InputStream is = Files.newInputStream(emptyFile)) {
```

Comment:
interesting corner case if you have an empty `ByteArrayInputStream` and you call `read` with a 0 size buffer it will return `EOF` BUT if you have an empty `FileInputStream` in that same scenario it will return 0.